<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SRTpipeline - Embedding and spatial clustering for single sample • SRTpipeline</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SRTpipeline - Embedding and spatial clustering for single sample">
<meta property="og:description" content="SRTpipeline">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SRTpipeline</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../articles/Installation.html">
    <span class="fas fa-book"></span>
     
    Install
  </a>
</li>
<li>
  <a href="../articles/get_started.html">
    <span class="fas fa-book"></span>
     
    Get Started
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book"></span>
     
    Single Data Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/single_basic_tutorial.html">Basic tutorial</a>
    </li>
    <li>
      <a href="../articles/dlpfc_tutorial.html">Embedding and clustering introduction</a>
    </li>
    <li>
      <a href="../articles/single_Embryo_SC_tutorial.html">SRT data clustering</a>
    </li>
    <li>
      <a href="../articles/single_Embryo_DRSC_tutorial.html">SRT data embedding and clustering</a>
    </li>
    <li>
      <a href="../articles/single_MOB_Anno_tutorial.html">SRT data embedding and annotation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book"></span>
     
    Multiple Data Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/multibatch_basic_tutorial.html">Basic tutorial</a>
    </li>
    <li>
      <a href="../articles/integration_introduction.html">Basic tutorial v2</a>
    </li>
    <li>
      <a href="../articles/multibatch_brain12_tutorial.html">SRT data clustering and integration</a>
    </li>
    <li>
      <a href="../articles/multibatch_HCC4_tutorial.html">SRT data embedding, clustering and integration example 1</a>
    </li>
    <li>
      <a href="../articles/multibatch_liver8_tutorial.html">SRT data embedding, clustering and integration example 2</a>
    </li>
    <li>
      <a href="../articles/multibatch_MOB16_tutorial.html">SRT data embedding, clustering and integration example 3</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/faq.html">
    <span class="fa fa-question-circle-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SRTpipeline - Embedding and spatial clustering for single sample</h1>
            
            <h4 data-toc-skip class="date">Compiled: 十二月 06, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/feiyoung/SRTpipeline/blob/HEAD/vignettes/dlpfc_tutorial.Rmd"><code>vignettes/dlpfc_tutorial.Rmd</code></a></small>
      <div class="hidden name"><code>dlpfc_tutorial.Rmd</code></div>

    </div>

    
    
<hr>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>For this tutorial, we will introduce how to use DR-SC model for joint dimension reduction and spaital clustering based on the R package <code>DR.SC</code>. Here, we will be analyzing a spatial transcriptomics dataset (SampleID: 151672) for human dorsolateral prefrontal cortex (DLPFC). There are 4015 spots and 33538 genes that were sequenced on the 10x Visium platform. Our preprocessed data can be downloaded <a href="https://github.com/feiyoung/PRECAST/blob/main/vignettes_data/dlpfc_151672.rda?raw=true" class="external-link">here</a>, and the raw data can be found <a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">here</a>.</p>
<div class="section level3">
<h3 id="prepare-srt-data-for-dr-sc-model">Prepare SRT data for DR-SC model<a class="anchor" aria-label="anchor" href="#prepare-srt-data-for-dr-sc-model"></a>
</h3>
<p>First, we loaded the required packages with the command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/feiyoung/SRTpipeline">SRTpipeline</a></span><span class="op">)</span></code></pre></div>
<p>Next, we download the data to the current working path for the followed analysis by the following command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/feiyoung/PRECAST/blob/main/vignettes_data/dlpfc_151672.rda?raw=true"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="st">"dlpfc_151672.rda"</span>, mode <span class="op">=</span> <span class="st">"wb"</span><span class="op">)</span></code></pre></div>
<p>Then load to R. The data is saved in a Seurat object named <code>dlpfc_151672</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"dlpfc_151672.rda"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pre-processing-workflow">Pre-processing workflow<a class="anchor" aria-label="anchor" href="#pre-processing-workflow"></a>
</h3>
<div class="section level4">
<h4 id="create-a-srtproject-object">Create a SRTProject object<a class="anchor" aria-label="anchor" href="#create-a-srtproject-object"></a>
</h4>
<p>First, we show how to create a SRTProject object step by step.</p>
<p>Because the gene name is ensembl in this data, we change the genes’ ensembl name to symbols for convenience of followed analyses. We can see most of ensembl names have been transferred to the symbol names and the genes not transferred are generally not in the variable gene set. Thhus, it does not matter for analyses that there are genes not be transferred.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="va">dlpfc_151672</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">counts</span>
<span class="co">## Use eg.db database: this method is fast</span>
<span class="va">symbol_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transferGeneNames.html">transferGeneNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span>, now_name <span class="op">=</span> <span class="st">"ensembl"</span>, to_name <span class="op">=</span> <span class="st">"symbol"</span>,
    species <span class="op">=</span> <span class="st">"Human"</span>, Method <span class="op">=</span> <span class="st">"eg.db"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">symbol_name</span>
<span class="va">symbol_name</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## create count matrix list: note each component has a name, i.e., `ID151672`.</span>

<span class="va">cntList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>ID151672 <span class="op">=</span> <span class="va">count_matrix</span><span class="op">)</span>

<span class="co">## create spatial coordinate matrix</span>
<span class="va">coordList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">dlpfc_151672</span><span class="op">$</span><span class="va">row</span>, <span class="va">dlpfc_151672</span><span class="op">$</span><span class="va">col</span><span class="op">)</span><span class="op">)</span>

<span class="co">## create metadata list</span>
<span class="va">meta.data</span> <span class="op">&lt;-</span> <span class="va">dlpfc_151672</span><span class="op">@</span><span class="va">meta.data</span>
<span class="va">metadataList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">meta.data</span><span class="op">)</span>

<span class="co">## create meta data for each data batches. Here we only have one data batch.</span>
<span class="va">sampleMetadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Human"</span>, tissues <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DLPFC"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">sampleMetadata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cntList</span><span class="op">)</span>

<span class="co">## Name of this project</span>
<span class="va">projectName</span> <span class="op">&lt;-</span> <span class="st">"Brain151672"</span>

<span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">dlpfc_151672</span><span class="op">)</span></code></pre></div>
<p>Next, we start creating a SRTProject object. We can print the basic information of this object, including three parts. The first part have the class of this object, outputPath of data that require to output, h5filePath that save the memory-cusuming data (i.e., count, logcount, …). The second part is about the datasets basic information, such as how many data batches(sample) and the data names, sample meta data (sampleColData) and meta data for each spot (cellMetaData). The last part is about downstream analyses information that is empty when this object created. Quality control is also performed when creating the SRTProject. SRTpipeline allows you to easily explore QC metrics and filter cells based on any user-defined criteria. A few QC metrics <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4758103/" class="external-link">commonly used</a> by the community include * The number of unique genes detected in each cell. + Low-quality cells or empty droplets will often have very few genes + Cell doublets or multiplets may exhibit an aberrantly high gene count * Similarly, the total number of molecules detected within a cell (correlates strongly with unique genes)</p>
<p>In the example below, we visualize QC metrics, and use these to filter cells. * filter out the spots with nonzero expression levels for less than 20 genes. * filter out genes with non-zero expression levels for fewer than 20 spots</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CreateSRTProject.html">CreateSRTProject</a></span><span class="op">(</span><span class="va">cntList</span>, <span class="va">coordList</span>, projectName <span class="op">=</span> <span class="va">projectName</span>, <span class="va">metadataList</span>, <span class="va">sampleMetadata</span>,
    min.spots <span class="op">=</span> <span class="fl">20</span>, min.genes <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="va">SRTProj</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="normalizing-the-data">Normalizing the data<a class="anchor" aria-label="anchor" href="#normalizing-the-data"></a>
</h4>
<p>After creating the SRTProject object, the next step is to normalize the data. Normalized values are stored in the h5 file of h5file path.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeSRT.html">normalizeSRT</a></span><span class="op">(</span><span class="va">SRTProj</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="feature-selection">Feature selection<a class="anchor" aria-label="anchor" href="#feature-selection"></a>
</h4>
<p>We next select a subset of genes that exhibit high spot-to-spot variation in the dataset (i.e, they are highly expressed in some spots, and lowly expressed in others). It has been found that focusing on these genes in downstream analysis helps to highlight biological signal in single-cell datasets <a href="https://www.nature.com/articles/nmeth.2645" class="external-link">here</a>.</p>
<div class="section level5">
<h5 id="choose-spatially-variable-features">Choose spatially variable features<a class="anchor" aria-label="anchor" href="#choose-spatially-variable-features"></a>
</h5>
<p>In the single_basic_tutorial, we introduced using <code>vst</code> to choose the highly vairiable genes. Here, we introduce to apply the SPARK-X to choose the top spatially variable genes (SVGs). SPARK-X is embeded in the <code>DR.SC</code> R package and is implement in the <code><a href="https://rdrr.io/pkg/DR.SC/man/FindSVGs.html" class="external-link">FindSVGs()</a></code> function. By default, we return 2,000 genes per dataset. These will be used in downstream analysis, like probabilistic embeddings and spatial clustering. To further speed up the computation, we first use <code>FindVariableFeatures()</code> to select 5,000 highly variable genes, then use SPARK-X to choose top SVGs in the implementation of <code><a href="https://rdrr.io/pkg/DR.SC/man/FindSVGs.html" class="external-link">FindSVGs()</a></code> function. The selected SVGs are saved in <code>SRTProj@geneMetaData$isVGs#combine</code> for followed analysse. In addition, the other columns of <code>SRTProj@geneMetaData</code> save the selected SVGs for each data batch. Because there is only one data batch, <code>SRTProj@geneMetaData$isSVGs#ID151672</code> is equal to <code>SRTProj@geneMetaData$isVGs#combine</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectVariableFeatures.html">selectVariableFeatures</a></span><span class="op">(</span><span class="va">SRTProj</span>, type <span class="op">=</span> <span class="st">"SVGs"</span>, method <span class="op">=</span> <span class="st">"SPARK-X"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">SRTProj</span><span class="op">@</span><span class="va">geneMetaData</span><span class="op">)</span></code></pre></div>
<p>Users can access the top SVGs using</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">svg4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/topVGs.html">topVGs</a></span><span class="op">(</span><span class="va">SRTProj</span>, ntop <span class="op">=</span> <span class="fl">4</span>, type <span class="op">=</span> <span class="st">"SVGs"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Visualize</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/EachExprSpaHeatMap.html">EachExprSpaHeatMap</a></span><span class="op">(</span><span class="va">SRTProj</span>, features <span class="op">=</span> <span class="va">svg4</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level5">
<h5 id="choose-highly-variable-features">Choose highly variable features<a class="anchor" aria-label="anchor" href="#choose-highly-variable-features"></a>
</h5>
<p>We can also directly use <code>FindVariableFeatures()</code> to choose the top highly variable genes (HVGs). We found the performance of using SVGs or HVGs has no significant difference for the downstream analyses, such as dimension reduction and spatial clustering.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># not run here SRTProj &lt;- selectVariableFeatures(SRTProj, type = 'HVGs', method='vst')</span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="calculate-the-adjcence-matrix">Calculate the adjcence matrix<a class="anchor" aria-label="anchor" href="#calculate-the-adjcence-matrix"></a>
</h4>
<p>Next, we define the neighbors for each spot by the spatial coordinates, which is represented by a sparse adjacency matrix (AdjMat) saved in the h5file. We can check the saved sparse adjacency matrix using <code>h5ls()</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Obtain adjacence matrix</span>
<span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddAdj.html">AddAdj</a></span><span class="op">(</span><span class="va">SRTProj</span>, platform <span class="op">=</span> <span class="st">"Visium"</span><span class="op">)</span>
<span class="fu">h5ls</span><span class="op">(</span><span class="va">SRTProj</span><span class="op">@</span><span class="va">projectMetadata</span><span class="op">$</span><span class="va">h5filePath</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="joint-dimension-reduction-and-spatial-clustering-using-dr-sc-model">Joint dimension reduction and spatial clustering using DR-SC model<a class="anchor" aria-label="anchor" href="#joint-dimension-reduction-and-spatial-clustering-using-dr-sc-model"></a>
</h2>
<p>We fit the DR-SC model by using the top 2000 SVGs. Users can also use the top HVGs since we found the performance of using SVGs or HVGs has no significant difference for the downstream analyses, such as dimension reduction and spatial clustering. Here, we specify the number of clusters as 5, while <code>DR.SC</code> can also select the number of clusters given a vector for argument <span class="math inline">\(K\)</span>.</p>
<p>Afeter finishing joint dimension reduction and spatial clustering, the SRTProject <code>SRTProj</code> has two more downstream analyses information: the low-dimensional embeddings DR.SC and inferred cluster labels, and they are saved in <code>SRTProj@reductions$DR.SC</code> and <code>SRTProj@clusters</code>, respectively.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Given K</span>
<span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Cluster_DRSC.html">Cluster_DRSC</a></span><span class="op">(</span><span class="va">SRTProj</span>, K <span class="op">=</span> <span class="fl">5</span>, q <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>
<span class="va">SRTProj</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p><code>SRTpipeline</code> package oﬀers a variety of visualization tools. To run tSNE in SRTpipeline we use the <code><a href="../reference/AddTSNE.html">AddTSNE()</a></code> function:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddTSNE.html">AddTSNE</a></span><span class="op">(</span><span class="va">SRTProj</span>, n_comp <span class="op">=</span> <span class="fl">2</span>, reduction <span class="op">=</span> <span class="st">"DR.SC"</span><span class="op">)</span></code></pre></div>
<p>The <code>reduction</code> tells <code><a href="../reference/AddTSNE.html">AddTSNE()</a></code> function uses specified reduction in the <code>SRTProj@reductions</code> slot.</p>
<p>To plot the two-dimensional tSNE results, we use the <code><a href="../reference/EmbedPlot.html">EmbedPlot()</a></code> function and pass the name of the tSNE embedding we just generated (“UMAP”). We can tell SRTpipeline how to color the spots by using <code>item</code> argument which tells SRTpipeline which data to use to find the specified metadata column provided to name.</p>
<p>We visualized the inferred domain types on embeddings using two components from tSNE. The tSNE plot showed the domain clusters were well segregated.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1">cols_cluster &lt;-<span class="st"> </span><span class="kw">chooseColors</span>(<span class="dt">n_colors =</span> <span class="dv">5</span>)</a>
<a class="sourceLine" id="cb15-2" title="2">p_tsne2 &lt;-<span class="st"> </span><span class="kw">EmbedPlot</span>(SRTProj, <span class="dt">item=</span><span class="st">'cluster'</span>, <span class="dt">plotEmbeddings =</span> <span class="st">'tSNE'</span>, <span class="dt">cols=</span>cols_cluster, <span class="dt">legend.position=</span><span class="st">'right'</span>)</a>
<a class="sourceLine" id="cb15-3" title="3">p_tsne2</a>
<a class="sourceLine" id="cb15-4" title="4"><span class="st">``</span></a>
<a class="sourceLine" id="cb15-5" title="5"></a>
<a class="sourceLine" id="cb15-6" title="6">To run UMAP <span class="cf">in</span> SRTpipeline we use the<span class="st">`</span><span class="dt"> AddUMAP()</span><span class="st">`</span> function. Frist, we evaluate the two<span class="op">-</span>dimensional UMAPs.</a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddUMAP.html">AddUMAP</a></span><span class="op">(</span><span class="va">SRTProj</span>, n_comp <span class="op">=</span> <span class="fl">2</span>, reduction <span class="op">=</span> <span class="st">"DR.SC"</span><span class="op">)</span>
<span class="va">p_umap2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EmbedPlot.html">EmbedPlot</a></span><span class="op">(</span><span class="va">SRTProj</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, plotEmbeddings <span class="op">=</span> <span class="st">"UMAP"</span>, cols <span class="op">=</span> <span class="va">cols_cluster</span>, legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>
<span class="va">p_umap2</span></code></pre></div>
<p>Next, we merge all plots into one figure.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drawFigs.html">drawFigs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">p_tsne2</span>, <span class="va">p_umap2</span><span class="op">)</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, common.legend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Calculate the 3-dimensional tSNEs based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddTSNE.html">AddTSNE</a></span><span class="op">(</span><span class="va">SRTProj</span>, n_comp <span class="op">=</span> <span class="fl">3</span>, reduction <span class="op">=</span> <span class="st">"DR.SC"</span><span class="op">)</span></code></pre></div>
<p>We visualize the 3-dimensional tSNEs on the spatial coordinates using RGB colors. We can see there are apparent spatial pattern for the embeddings. Users can also plot the 3-dimensional UMAP RGB plot similarly.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/EachRGBSpaHeatMap.html">EachRGBSpaHeatMap</a></span><span class="op">(</span><span class="va">SRTProj</span>, plot_type <span class="op">=</span> <span class="st">"tSNE"</span>, title_name <span class="op">=</span> <span class="st">"tSNE RGB plot: "</span><span class="op">)</span></code></pre></div>
<p>Then, we visualize the spatial distribution of cluster labels that shows the layer structure.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/EachClusterSpaHeatMap.html">EachClusterSpaHeatMap</a></span><span class="op">(</span><span class="va">SRTProj</span>, cols <span class="op">=</span> <span class="va">cols_cluster</span>, legend.position <span class="op">=</span> <span class="st">"right"</span>, base_size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>
<span class="co"># remove the border EachClusterSpaHeatMap(SRTProj, cols=cols_cluster,</span>
<span class="co"># legend.position='right',base_size=16, border_col='white')</span></code></pre></div>
<p>We plotted the heatmap of Pearson’s correlation coefcients of the DR.SC embeddings among the detected domains shows the good separation of the estimated embeddings across domains and the correlations between deeper layers were high, e.g., there were high correlations between domain 2 and 3, while correlations among the separated layers were low, i.e., domain 1 and 4.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/CCHeatMap.html">CCHeatMap</a></span><span class="op">(</span><span class="va">SRTProj</span>, reduction <span class="op">=</span> <span class="st">"DR.SC"</span>, grp_color <span class="op">=</span> <span class="va">cols_cluster</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="deg-analysis">DEG analysis<a class="anchor" aria-label="anchor" href="#deg-analysis"></a>
</h2>
<p>To do downstream analyses, we require to get the count data from h5file. The function <code><a href="../reference/getGeneSpotData.html">getGeneSpotData()</a></code> access the gene-by-spot count matrix and other data in <code>SRTProj</code> and then return a <code>SpatialExperiment</code> object, including two assays: counts, logcounts; rowData: from <code>SRTProj@geneMetaData</code>; colData: from <code>SRTProj@cellMetaData</code>, <code>SRTProj@clusters</code> and sample_id; reducedDim: from <code>SRTProj@reductions</code>, <code>SRTProj@plotEmbeddings</code> and sptial coordinates <code>Coord</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGeneSpotData.html">getGeneSpotData</a></span><span class="op">(</span><span class="va">SRTProj</span><span class="op">)</span>
<span class="va">spe</span></code></pre></div>
<p>After fitting DR-SC model, we can perform differentially expression analysis using the spatial cluster labels. First, we detect the DE genes for cluster 1, where the argument <code>use_cluster</code> tells <code><a href="../reference/FindDEGs.html">FindDEGs()</a></code> uses which colname name in the <code>colData(spe)</code> to do DEG analysis.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat_deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindDEGs.html">FindDEGs</a></span><span class="op">(</span><span class="va">spe</span>, cluster1 <span class="op">=</span> <span class="st">"1"</span>, use_cluster <span class="op">=</span> <span class="st">"clusters"</span><span class="op">)</span>
<span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Seurat-methods.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">dat_deg</span>, <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">0.25</span><span class="op">)</span></code></pre></div>
<p>We perform differential expression analysis for all clusters by using <code><a href="../reference/FindAllDEGs.html">FindAllDEGs()</a></code> function, then the DE genes’ information is saved in a <code>DataFrame</code> object <code>dat_degs</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat_degs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindAllDEGs.html">FindAllDEGs</a></span><span class="op">(</span><span class="va">spe</span>, use_cluster <span class="op">=</span> <span class="st">"clusters"</span><span class="op">)</span>
<span class="va">dat_degs</span></code></pre></div>
<p>We identify the significant DE genes by two criteria: (a) adjustd p-value less than 0.01 and (b) average log fold change greater than 0.4.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">degs_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Seurat-methods.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">dat_degs</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.01</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">0.4</span><span class="op">)</span>
<span class="va">degs_sig</span></code></pre></div>
<p>In the following, we perform gene set enrichment analysis for the DE genes of each Domain identified by DR-SC model using R package <code>gprofiler2</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gprofiler2</span><span class="op">)</span>
<span class="va">termList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">k</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span> <span class="op">{</span>
    <span class="co"># k &lt;- 1</span>
    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"k = "</span>, <span class="va">k</span>, <span class="st">"\n"</span><span class="op">)</span>
    <span class="va">dat_degs_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Seurat-methods.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">degs_sig</span>, <span class="va">cluster</span> <span class="op">==</span> <span class="va">k</span><span class="op">)</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">dat_degs_sub</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span>
        <span class="kw">next</span>
    <span class="va">que1</span> <span class="op">&lt;-</span> <span class="va">dat_degs_sub</span><span class="op">$</span><span class="va">gene</span>
    <span class="va">gostres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gost</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">que1</span>, organism <span class="op">=</span> <span class="st">"hsapiens"</span>, correction_method <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span>
    <span class="va">termList</span><span class="op">[[</span><span class="va">k</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">gostres</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">termList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">result</span><span class="op">)</span></code></pre></div>
<p>To understand the functions of the identified spatial domains by DR-SC model, we compare the top significant biological process (BP) pathways in GO database for the DE genes from Domain 1 and 3. Here, we only show to visualize the significant BP pathways and users can explore the other databases such as KEGG and HPA.</p>
<p>The enrichment analysis shows that genes specific to Domain 1 were highly enriched for galactosylceramide metabolic process, galactosylceramide biosynthetic process and galactolipid metabolic process, which may indicate the cells in Domain 1 play an important role to achieve these functions. The genes specific to Domain 4 were highly enriched for negative regulation of response to external stimulus and negative regulation of response to extracellular stimulus.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Most commonly used databases</span>
<span class="va">source_set</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:BP"</span>, <span class="st">"GO:CC"</span>, <span class="st">"GO:MF"</span>, <span class="st">"KEGG"</span>, <span class="st">"HPA"</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"steelblue3"</span>, <span class="st">"goldenrod"</span>, <span class="st">"brown3"</span>, <span class="st">"#f98866"</span>, <span class="st">"#CE6DBD"</span><span class="op">)</span>
<span class="co">## Here, we show GO:BP</span>
<span class="va">source1</span> <span class="op">&lt;-</span> <span class="st">"GO:BP"</span>
<span class="va">ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">source_set</span> <span class="op">==</span> <span class="va">source1</span><span class="op">)</span>
<span class="va">ntop</span> <span class="op">=</span> <span class="fl">5</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cols</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">source_set</span>
<span class="va">pList_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">ii</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span> <span class="op">{</span>
    <span class="co">## ii &lt;- 4</span>
    <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"ii="</span>, <span class="va">ii</span><span class="op">)</span>
    <span class="va">dat</span> <span class="op">&lt;-</span> <span class="va">termList</span><span class="op">[[</span><span class="va">ii</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">result</span>
    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
        <span class="va">dat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Seurat-methods.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">dat</span>, <span class="va">term_size</span> <span class="op">&lt;</span> <span class="fl">500</span><span class="op">)</span>
        <span class="va">dat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_top_pathway.html">get_top_pathway</a></span><span class="op">(</span><span class="va">dat1</span>, ntop <span class="op">=</span> <span class="va">ntop</span>, source_set <span class="op">=</span> <span class="va">source1</span><span class="op">)</span>
        <span class="va">dat1</span> <span class="op">&lt;-</span> <span class="va">dat1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">)</span>, <span class="op">]</span>
        <span class="va">dat1</span><span class="op">$</span><span class="va">nlog10P</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">$</span><span class="va">p_value</span><span class="op">)</span>

        <span class="va">pList_enrich</span><span class="op">[[</span><span class="va">ii</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/barPlot_enrich.html">barPlot_enrich</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">$</span><span class="va">nlog10P</span><span class="op">)</span>, <span class="op">]</span>, source <span class="op">=</span> <span class="st">"source"</span>, <span class="st">"term_name"</span>,
            <span class="st">"nlog10P"</span>, cols <span class="op">=</span> <span class="va">cols</span><span class="op">[</span><span class="va">source_set</span><span class="op">[</span><span class="va">ss</span><span class="op">]</span><span class="op">]</span>, base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"-log10(p-adj)"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">xlab</span><span class="op">(</span><span class="st">"Biological terms"</span><span class="op">)</span> <span class="op">+</span>
            <span class="fu">ggtitle</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Domain"</span>, <span class="va">ii</span><span class="op">)</span><span class="op">)</span>
    <span class="op">}</span>

<span class="op">}</span>

<span class="fu"><a href="../reference/drawFigs.html">drawFigs</a></span><span class="op">(</span><span class="va">pList_enrich</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">4</span><span class="op">)</span><span class="op">]</span>, common.legend <span class="op">=</span> <span class="cn">T</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span>, align <span class="op">=</span> <span class="st">"hv"</span><span class="op">)</span></code></pre></div>
<p>We take out the top DE genes for each cluster for visualization.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">dat_degs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="va">as.data.frame</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="va">n</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">topGene</span>
<span class="va">topGene</span></code></pre></div>
<p>Then we plot the volin plot of logcount expressions grouped by clusters, which suggests the marker gene has higher expression in the corresponding cluster.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html" class="external-link">plotExpression</a></span><span class="op">(</span><span class="va">spe</span>, features <span class="op">=</span> <span class="va">topGene</span><span class="op">$</span><span class="va">gene</span>, x <span class="op">=</span> <span class="st">"clusters"</span>, exprs_values <span class="op">=</span> <span class="st">"logcounts"</span>, colour_by <span class="op">=</span> <span class="st">"clusters"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="trajectory-inference">Trajectory inference<a class="anchor" aria-label="anchor" href="#trajectory-inference"></a>
</h2>
<p>Next, we performed trajectory inference using the embeddings and domain labels jointly estimated by DR.SC model. The <code><a href="../reference/EmbedPlot.html">EmbedPlot()</a></code> function can be used to visualize the inferred pseudotime on a specified embedding. If there is only one data batch, the function <code><a href="../reference/EachEmbedPlot.html">EachEmbedPlot()</a></code> plays the same role as function <code><a href="../reference/EmbedPlot.html">EmbedPlot()</a></code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddTrajectory.html">AddTrajectory</a></span><span class="op">(</span><span class="va">spe</span>, reduction <span class="op">=</span> <span class="st">"DR.SC"</span>, name <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EmbedPlot.html">EmbedPlot</a></span><span class="op">(</span><span class="va">spe</span>, reduction <span class="op">=</span> <span class="st">"DR.SC"</span>, colour_by <span class="op">=</span> <span class="st">"PT"</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EmbedPlot.html">EmbedPlot</a></span><span class="op">(</span><span class="va">spe</span>, reduction <span class="op">=</span> <span class="st">"Coord"</span>, colour_by <span class="op">=</span> <span class="st">"PT"</span><span class="op">)</span>
<span class="fu"><a href="../reference/drawFigs.html">drawFigs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span><span class="op">)</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, common.legend <span class="op">=</span> <span class="cn">TRUE</span>, legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>

<span class="co"># save(spe, SRTProj, file=paste0(SRTProj@projectMetadata$outputPath,'/SRTProj.rds'))</span>
<span class="co"># load('F:/Research</span>
<span class="co"># paper/IntegrateDRcluster/AnalysisCode/SRTpipeline/vignettes/DLPFC151672/SRTProj.rds')</span></code></pre></div>
<details><p><summary><strong>Session Info</strong></summary></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">## Running under: Windows 10 x64 (build 22000)</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] LC_COLLATE=Chinese (Simplified)_China.936 </span>
<span class="co">## [2] LC_CTYPE=Chinese (Simplified)_China.936   </span>
<span class="co">## [3] LC_MONETARY=Chinese (Simplified)_China.936</span>
<span class="co">## [4] LC_NUMERIC=C                              </span>
<span class="co">## [5] LC_TIME=Chinese (Simplified)_China.936    </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] SRTpipeline_0.1.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                  reticulate_1.25            </span>
<span class="co">##   [3] tidyselect_1.1.2            htmlwidgets_1.5.4          </span>
<span class="co">##   [5] grid_4.1.2                  BiocParallel_1.28.3        </span>
<span class="co">##   [7] Rtsne_0.16                  munsell_0.5.0              </span>
<span class="co">##   [9] ScaledMatrix_1.2.0          codetools_0.2-18           </span>
<span class="co">##  [11] ragg_1.2.2                  ica_1.0-2                  </span>
<span class="co">##  [13] future_1.26.1               miniUI_0.1.1.1             </span>
<span class="co">##  [15] spatstat.random_2.2-0       colorspace_2.0-3           </span>
<span class="co">##  [17] progressr_0.10.1            Biobase_2.54.0             </span>
<span class="co">##  [19] knitr_1.37                  rstudioapi_0.13            </span>
<span class="co">##  [21] Seurat_4.1.1                stats4_4.1.2               </span>
<span class="co">##  [23] SingleCellExperiment_1.16.0 ROCR_1.0-11                </span>
<span class="co">##  [25] ggsignif_0.6.3              tensor_1.5                 </span>
<span class="co">##  [27] listenv_0.8.0               MatrixGenerics_1.6.0       </span>
<span class="co">##  [29] GenomeInfoDbData_1.2.7      polyclip_1.10-0            </span>
<span class="co">##  [31] rprojroot_2.0.3             parallelly_1.32.0          </span>
<span class="co">##  [33] vctrs_0.4.1                 generics_0.1.2             </span>
<span class="co">##  [35] xfun_0.29                   ggthemes_4.2.4             </span>
<span class="co">##  [37] R6_2.5.1                    GenomeInfoDb_1.30.1        </span>
<span class="co">##  [39] ggbeeswarm_0.6.0            rsvd_1.0.5                 </span>
<span class="co">##  [41] bitops_1.0-7                spatstat.utils_3.0-1       </span>
<span class="co">##  [43] cachem_1.0.6                DelayedArray_0.20.0        </span>
<span class="co">##  [45] assertthat_0.2.1            promises_1.2.0.1           </span>
<span class="co">##  [47] scales_1.2.0                beeswarm_0.4.0             </span>
<span class="co">##  [49] rgeos_0.5-9                 gtable_0.3.0               </span>
<span class="co">##  [51] beachmat_2.10.0             globals_0.15.0             </span>
<span class="co">##  [53] goftest_1.2-3               rlang_1.0.2                </span>
<span class="co">##  [55] systemfonts_1.0.4           splines_4.1.2              </span>
<span class="co">##  [57] rstatix_0.7.0               lazyeval_0.2.2             </span>
<span class="co">##  [59] broom_0.7.12                spatstat.geom_2.4-0        </span>
<span class="co">##  [61] yaml_2.3.6                  reshape2_1.4.4             </span>
<span class="co">##  [63] abind_1.4-5                 backports_1.4.1            </span>
<span class="co">##  [65] httpuv_1.6.5                tools_4.1.2                </span>
<span class="co">##  [67] ggplot2_3.3.6               ellipsis_0.3.2             </span>
<span class="co">##  [69] spatstat.core_2.4-4         jquerylib_0.1.4            </span>
<span class="co">##  [71] RColorBrewer_1.1-3          BiocGenerics_0.40.0        </span>
<span class="co">##  [73] ggridges_0.5.3              Rcpp_1.0.8.3               </span>
<span class="co">##  [75] plyr_1.8.7                  sparseMatrixStats_1.6.0    </span>
<span class="co">##  [77] zlibbioc_1.40.0             purrr_0.3.4                </span>
<span class="co">##  [79] RCurl_1.98-1.6              ggpubr_0.4.0               </span>
<span class="co">##  [81] rpart_4.1.16                deldir_1.0-6               </span>
<span class="co">##  [83] viridis_0.6.2               pbapply_1.5-0              </span>
<span class="co">##  [85] cowplot_1.1.1               S4Vectors_0.32.3           </span>
<span class="co">##  [87] zoo_1.8-10                  SeuratObject_4.1.0         </span>
<span class="co">##  [89] SummarizedExperiment_1.24.0 ggrepel_0.9.1              </span>
<span class="co">##  [91] cluster_2.1.2               fs_1.5.2                   </span>
<span class="co">##  [93] magrittr_2.0.3              GiRaF_1.0.1                </span>
<span class="co">##  [95] data.table_1.14.2           scattermore_0.8            </span>
<span class="co">##  [97] lmtest_0.9-40               RANN_2.6.1                 </span>
<span class="co">##  [99] fitdistrplus_1.1-8          matrixStats_0.62.0         </span>
<span class="co">## [101] PRECAST_1.3                 patchwork_1.1.1            </span>
<span class="co">## [103] mime_0.12                   evaluate_0.15              </span>
<span class="co">## [105] xtable_1.8-4                mclust_5.4.10              </span>
<span class="co">## [107] IRanges_2.28.0              gridExtra_2.3              </span>
<span class="co">## [109] compiler_4.1.2              scater_1.25.1              </span>
<span class="co">## [111] tibble_3.1.7                KernSmooth_2.23-20         </span>
<span class="co">## [113] crayon_1.5.1                htmltools_0.5.2            </span>
<span class="co">## [115] mgcv_1.8-39                 later_1.3.0                </span>
<span class="co">## [117] tidyr_1.2.0                 DBI_1.1.2                  </span>
<span class="co">## [119] formatR_1.11                MASS_7.3-55                </span>
<span class="co">## [121] car_3.0-12                  Matrix_1.4-0               </span>
<span class="co">## [123] cli_3.2.0                   parallel_4.1.2             </span>
<span class="co">## [125] igraph_1.3.2                DR.SC_3.0                  </span>
<span class="co">## [127] GenomicRanges_1.46.1        pkgconfig_2.0.3            </span>
<span class="co">## [129] pkgdown_2.0.6               sp_1.5-0                   </span>
<span class="co">## [131] plotly_4.10.0               scuttle_1.4.0              </span>
<span class="co">## [133] spatstat.sparse_2.1-1       vipor_0.4.5                </span>
<span class="co">## [135] bslib_0.3.1                 XVector_0.34.0             </span>
<span class="co">## [137] CompQuadForm_1.4.3          stringr_1.4.0              </span>
<span class="co">## [139] digest_0.6.29               sctransform_0.3.3          </span>
<span class="co">## [141] RcppAnnoy_0.0.19            spatstat.data_3.0-0        </span>
<span class="co">## [143] rmarkdown_2.11              leiden_0.4.2               </span>
<span class="co">## [145] uwot_0.1.11                 DelayedMatrixStats_1.16.0  </span>
<span class="co">## [147] shiny_1.7.1                 gtools_3.9.2.2             </span>
<span class="co">## [149] lifecycle_1.0.1             nlme_3.1-155               </span>
<span class="co">## [151] jsonlite_1.8.0              carData_3.0-5              </span>
<span class="co">## [153] BiocNeighbors_1.12.0        desc_1.4.0                 </span>
<span class="co">## [155] viridisLite_0.4.0           fansi_1.0.3                </span>
<span class="co">## [157] pillar_1.7.0                lattice_0.20-45            </span>
<span class="co">## [159] fastmap_1.1.0               httr_1.4.3                 </span>
<span class="co">## [161] survival_3.2-13             glue_1.6.2                 </span>
<span class="co">## [163] png_0.1-7                   stringi_1.7.6              </span>
<span class="co">## [165] sass_0.4.1                  textshaping_0.3.6          </span>
<span class="co">## [167] BiocSingular_1.10.0         memoise_2.0.1              </span>
<span class="co">## [169] dplyr_1.0.9                 irlba_2.3.5                </span>
<span class="co">## [171] future.apply_1.9.0</span></code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
