<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started with SRTpipeline • SRTpipeline</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with SRTpipeline">
<meta property="og:description" content="SRTpipeline">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SRTpipeline</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../articles/Installation.html">
    <span class="fas fa-book"></span>
     
    Install
  </a>
</li>
<li>
  <a href="../articles/get_started.html">
    <span class="fas fa-book"></span>
     
    Get Started
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book"></span>
     
    Single Data Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/single_basic_tutorial.html">Basic tutorial</a>
    </li>
    <li>
      <a href="../articles/dlpfc_tutorial.html">Embedding and clustering introduction</a>
    </li>
    <li>
      <a href="../articles/single_Embryo_SC_tutorial.html">SRT data clustering</a>
    </li>
    <li>
      <a href="../articles/single_Embryo_DRSC_tutorial.html">SRT data embedding and clustering</a>
    </li>
    <li>
      <a href="../articles/single_MOB_Anno_tutorial.html">SRT data embedding and annotation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book"></span>
     
    Multiple Data Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/multibatch_basic_tutorial.html">Basic tutorial</a>
    </li>
    <li>
      <a href="../articles/integration_introduction.html">Basic tutorial v2</a>
    </li>
    <li>
      <a href="../articles/multibatch_brain12_tutorial.html">SRT data clustering and integration</a>
    </li>
    <li>
      <a href="../articles/multibatch_HCC4_tutorial.html">SRT data embedding, clustering and integration example 1</a>
    </li>
    <li>
      <a href="../articles/multibatch_liver8_tutorial.html">SRT data embedding, clustering and integration example 2</a>
    </li>
    <li>
      <a href="../articles/multibatch_MOB16_tutorial.html">SRT data embedding, clustering and integration example 3</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/faq.html">
    <span class="fa fa-question-circle-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="get_started_files/kePrint-0.0.1/kePrint.js"></script><link href="get_started_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started with SRTpipeline</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/feiyoung/SRTpipeline/blob/HEAD/vignettes/get_started.Rmd"><code>vignettes/get_started.Rmd</code></a></small>
      <div class="hidden name"><code>get_started.Rmd</code></div>

    </div>

    
    
<style>
.two-column-htmltable {
  width: 66.666%;
}
.one-column-htmltable {
  width: 33.333%;
}

</style>
<p>We provide a series of vignettes, tutorials, and analysis walkthroughs to help users get started with SRTpipeline. You can also check out our <a href="../reference/index.html">Reference page</a> which contains a full list of functions available to users.</p>
<div class="section level2">
<h2 id="introductory-vignettes">Introductory Vignettes<a class="anchor" aria-label="anchor" href="#introductory-vignettes"></a>
</h2>
<p>For new users of SRTpipeline, we suggest starting with a guided walk through of a spatial transcriptomics dataset (SampleID: 151672) for human dorsolateral prefrontal cortex (DLPFC) from <a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">Github</a>. This tutorial implements the major components of a standard unsupervised clustering workflow including QC and data filtration, calculation of high-variance genes or spatially variable genes, dimensional reduction, Markov random field-based spatial clustering, and the identification of cluster markers.</p>
<p>We provide additional introductory vignettes for users who are interested in analyzing multiple datasets.</p>
<table class="table gmisc_table" style="border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;"><tbody>
<tr style="border-top: 2px solid grey;">
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="single_basic_tutorial.html">A basic overview for single data analyses — taking DLPFC 151672 as an example</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="multibatch_basic_tutorial.html">A basic overview for multiple data analyses — taking eight Liver ST datasets an example</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="dlpfc_tutorial.html">Embedding and spatial clustering for single sample — DLPFC 151672</a>
</div>
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
A basic overview of SRTpipeline that includes an introduction to common analytical workflows.
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Learn to integrate spatially-resolved transcriptomic data with examples from 10x Visium.
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
An introduction to working with single SRT sample in SRTpipeline.
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="single_basic_tutorial.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="multibatch_basic_tutorial.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="dlpfc_tutorial.html" role="button" style="width:100%">GO</a>
</td>
</tr>
</tbody></table>
</div>
<div class="section level2">
<h2 id="single-srt-data-analyses">Single SRT Data Analyses<a class="anchor" aria-label="anchor" href="#single-srt-data-analyses"></a>
</h2>
<p>Recently, we have developed <a href="https://www.biorxiv.org/content/10.1101/2022.06.26.497672v2" class="external-link">computational methods</a> for integrated analysis of SRT datasets generated across different conditions, experimental batches, individuals or datasets with only partially shared cell/domain clusters. As an example, we provide a guided walk through for integrating and comparing hepatocellular carcinoma (HCC) Visium data generated under different conditions. There are four slides of in-house HCC data generated using the 10x Visium platform, with two slides from tumors (HCC1 and HCC2) and two from tumor-adjacent tissues (HCC3 and HCC4) from an HCC patient. We provide additional vignettes demonstrating how to leverage SC-MEB2 to efficiently integrate large datasets.</p>
<table class="table gmisc_table" style="border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;"><tbody>
<tr style="border-top: 2px solid grey;">
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="single_Embryo_SC_tutorial.html">SRT data clustering — Embryo seqFISH dataset</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="single_Embryo_DRSC_tutorial.html">SRT data embedding and clustering — Embryo seqFISH dataset</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="single_MOB_Anno_tutorial.html">SRT data embedding and annotation</a>
</div>
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
An introduction to SC-MEB
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Learn how to use DR-SC
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Learn how to use SpatailAnno
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="single_Embryo_SC_tutorial.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="single_Embryo_DRSC_tutorial.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="single_MOB_Anno_tutorial.html" role="button" style="width:100%">GO</a>
</td>
</tr>
</tbody></table>
</div>
<div class="section level2">
<h2 id="mulitple-srt-data-integration">Mulitple SRT Data Integration<a class="anchor" aria-label="anchor" href="#mulitple-srt-data-integration"></a>
</h2>
<p>Recently, we have developed <a href="https://www.biorxiv.org/content/10.1101/2022.06.26.497672v2" class="external-link">computational methods</a> for integrated analysis of SRT datasets generated across different conditions, experimental batches, individuals or datasets with only partially shared cell/domain clusters. As an example, we provide a guided walk through for integrating and comparing hepatocellular carcinoma (HCC) Visium data generated under different conditions. There are four slides of in-house HCC data generated using the 10x Visium platform, with two slides from tumors (HCC1 and HCC2) and two from tumor-adjacent tissues (HCC3 and HCC4) from an HCC patient. We provide additional vignettes demonstrating how to leverage SC-MEB2 to efficiently integrate large datasets.</p>
<table class="table gmisc_table" style="border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;"><tbody>
<tr style="border-top: 2px solid grey;">
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="integration_introduction.html">Introduction to SRT integration — eight Liver ST datasets</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="multibatch_brain12_tutorial.html">SRT clustering and integration based on uncorrected PCs — 12 DLPFC Visium datasets</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="multibatch_HCC4_tutorial.html">SRT embedding, clustering and integration based on normlized expression — four hepatocellular carcinoma Visium data</a>
</div>
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
An introduction to integrating spatially resolved transcriptomics (SRT) datasets in order to identify and compare shared embeddings and cell types across experiments.
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Learn how to rapidly integrate multiple SRT datasets, achieve spatial clustering and align embeddings using uncorrected PCs.
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Learn how to integrate multiple SRT datasets, achieve embeddings alignment and spatial clustering using normlized gene expression.
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="integration_introduction.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="multibatch_brain12_tutorial.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="multibatch_HCC4_tutorial.html" role="button" style="width:100%">GO</a>
</td>
</tr>
</tbody></table>
<table class="table gmisc_table" style="border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;"><tbody>
<tr style="border-top: 2px solid grey;">
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="multibatch_liver8_tutorial.html">SRT embedding, clustering and integration based on normlized expression — eight mouse liver ST data</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="multibatch_MOB16_tutorial.html">SRT embedding, clustering and integration based on normlized expression — 16 mouse olfactory bulb Slide-seqV2 data</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="integration_large_datasets.html">Tips for integrating large datasets</a>
</div>
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
Analyze query data in the context of multimodal reference atlases.
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Annotate, visualize, and interpret an scATAC-seq experiment using scRNA-seq data from the same biological system.
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Tips and examples for integrating very large scRNA-seq datasets (including &gt;200,000 cells).
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="multibatch_liver8_tutorial.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="multibatch_MOB16_tutorial.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="integration_large_datasets.html" role="button" style="width:100%">GO</a>
</td>
</tr>
</tbody></table>
</div>
<div class="section level2">
<h2 id="other">Other<a class="anchor" aria-label="anchor" href="#other"></a>
</h2>
<p>Here we provide a series of short vignettes to demonstrate a number of features that are commonly used in SRTpipeline. We’ve focused the vignettes around questions that we frequently receive from users. Click on a vignette to get started.</p>
<table class="table gmisc_table" style="border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;"><tbody>
<tr style="border-top: 2px solid grey;">
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="visualization_vignette.html">Visualization</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="cell_cycle_vignette.html">XX</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="de_vignette.html">XX</a>
</div>
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
An overview of the major visualization functionality within SRTpipeline.
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
XX examples
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
XX examples
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="visualization_vignette.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="cell_cycle_vignette.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="de_vignette.html" role="button" style="width:100%">GO</a>
</td>
</tr>
</tbody></table>
<table class="table gmisc_table" style="border-collapse: collapse; margin-top: 1em; margin-bottom: 1em;"><tbody>
<tr style="border-top: 2px solid grey;">
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="hashing_vignette.html">XX</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="conversion_vignette.html">Interoperability with Other Analysis Tools</a>
</div>
</td>
<td style="padding-left: .75em; width: 30%; border-top: 2px solid grey; border-right: 1px solid black; text-align: left;">
<div style="text-align:center; font-size:large; font-weight:bold">
<a href="future_vignette.html">Parallelization</a>
</div>
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
XX examples
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Convert data between formats for different analysis tools.
</td>
<td style="padding-left: .75em; width: 30%; border-right: 1px solid black; text-align: left;">
Speed up compute-intensive functions with parallelization.
</td>
</tr>
<tr>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-left: 1px solid black; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="hashing_vignette.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="conversion_vignette.html" role="button" style="width:100%">GO</a>
</td>
<td style="padding-left: .75em; width: 30%; border-bottom: 2px solid grey; border-right: 1px solid black; text-align: left;">
<a class="btn btn-primary" href="future_vignette.html" role="button" style="width:100%">GO</a>
</td>
</tr>
</tbody></table>
</div>
<div class="section level2">
<h2 id="srtpipelinewrappers">SRTpipelineWrappers<a class="anchor" aria-label="anchor" href="#srtpipelinewrappers"></a>
</h2>
<p>In order to facilitate the use of community tools with SRTpipeline, we provide the SRTpipeline Wrappers package, which contains code to run other analysis tools on SRTpipeline objects. For the initial release, we provide wrappers for a few packages in the table below but would encourage other package developers interested in interfacing with SRTpipeline to check out our contributor guide <a href="https://github.com/satijalab/SRTpipeline.wrappers/wiki/Submission-Process" class="external-link">here</a>.</p>
<table class="table table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
Package
</th>
<th style="text-align:left;">
Vignette
</th>
<th style="text-align:left;">
Reference
</th>
<th style="text-align:left;">
Source
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
alevin
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/alevin.html" class="external-link">Import alevin counts into SRTpipeline</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1186/s13059-019-1670-y" class="external-link">Srivastava et. al., Genome Biology 2019</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/k3yavi/alevin-Rtools" class="external-link">https://github.com/k3yavi/alevin-Rtools</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
ALRA
</td>
<td style="text-align:left;">
<a href="https://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline.wrappers/blob/master/docs/alra.html" class="external-link">Zero-preserving imputation with ALRA</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1101/397588" class="external-link">Linderman et al, bioRxiv 2018</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/KlugerLab/ALRA" class="external-link">https://github.com/KlugerLab/ALRA</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
CoGAPS
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/cogaps.html" class="external-link">Running CoGAPS on SRTpipeline Objects</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1016/j.cels.2019.04.004" class="external-link">Stein-O’Brien et al, Cell Systems 2019</a>
</td>
<td style="text-align:left;">
<a href="https://www.bioconductor.org/packages/release/bioc/html/CoGAPS.html" class="external-link">https://www.bioconductor.org/packages/release/bioc/html/CoGAPS.html</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
Conos
</td>
<td style="text-align:left;">
<a href="https://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline.wrappers/blob/master/docs/conos.html" class="external-link">Integration of datasets using Conos</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1038/s41592-019-0466-z" class="external-link">Barkas et al, Nature Methods 2019</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/hms-dbmi/conos" class="external-link">https://github.com/hms-dbmi/conos</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
fastMNN
</td>
<td style="text-align:left;">
<a href="https://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline.wrappers/blob/master/docs/fast_mnn.html" class="external-link">Running fastMNN on SRTpipeline Objects</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1038/nbt.4091" class="external-link">Haghverdi et al, Nature Biotechnology 2018</a>
</td>
<td style="text-align:left;">
<a href="https://bioconductor.org/packages/release/bioc/html/scran.html" class="external-link">https://bioconductor.org/packages/release/bioc/html/scran.html</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
glmpca
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/glmpca.html" class="external-link">Running GLM-PCA on a SRTpipeline Object</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1186/s13059-019-1861-6" class="external-link">Townes et al, Genome Biology 2019</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/willtownes/glmpca" class="external-link">https://github.com/willtownes/glmpca</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
Harmony
</td>
<td style="text-align:left;">
<a href="https://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline.wrappers/blob/master/docs/harmony.html" class="external-link">Integration of datasets using Harmony</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1038/s41592-019-0619-0" class="external-link">Korsunsky et al, Nature Methods 2019</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/immunogenomics/harmony" class="external-link">https://github.com/immunogenomics/harmony</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
LIGER
</td>
<td style="text-align:left;">
<a href="https://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline.wrappers/blob/master/docs/liger.html" class="external-link">Integrating SRTpipeline objects using LIGER</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1016/j.cell.2019.05.006" class="external-link">Welch et al, Cell 2019</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/MacoskoLab/liger" class="external-link">https://github.com/MacoskoLab/liger</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
Monocle3
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/monocle3.html" class="external-link">Calculating Trajectories with Monocle 3 and SRTpipeline</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1038/s41586-019-0969-x" class="external-link">Cao et al, Nature 2019</a>
</td>
<td style="text-align:left;">
<a href="https://cole-trapnell-lab.github.io/monocle3" class="external-link">https://cole-trapnell-lab.github.io/monocle3</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
Nebulosa
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/nebulosa.html" class="external-link">Visualization of gene expression with Nebulosa</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/powellgenomicslab/Nebulosa" class="external-link">Jose Alquicira-Hernandez and Joseph E. Powell, Under Review</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/powellgenomicslab/Nebulosa" class="external-link">https://github.com/powellgenomicslab/Nebulosa</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
schex
</td>
<td style="text-align:left;">
<a href="https://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline.wrappers/blob/master/docs/schex.html" class="external-link">Using schex with SRTpipeline</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/0.1242/dev.173807" class="external-link">Freytag, R package 2019</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/SaskiaFreytag/schex" class="external-link">https://github.com/SaskiaFreytag/schex</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
scVelo
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/scvelo.html" class="external-link">Estimating RNA Velocity using SRTpipeline and scVelo</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1101/820936" class="external-link">Bergen et al, bioRxiv 2019</a>
</td>
<td style="text-align:left;">
<a href="https://scvelo.readthedocs.io/" class="external-link">https://scvelo.readthedocs.io/</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
Velocity
</td>
<td style="text-align:left;">
<a href="https://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline.wrappers/blob/master/docs/velocity.html" class="external-link">Estimating RNA Velocity using SRTpipeline</a>
</td>
<td style="text-align:left;">
<a href="10.1038/s41586-018-0414-6">La Manno et al, Nature 2018</a>
</td>
<td style="text-align:left;">
<a href="https://velocyto.org" class="external-link">https://velocyto.org</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
CIPR
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/cipr.html" class="external-link">Using CIPR with human PBMC data</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1186/s12859-020-3538-2" class="external-link">Ekiz et. al., BMC Bioinformatics 2020</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/atakanekiz/CIPR-Package" class="external-link">https://github.com/atakanekiz/CIPR-Package</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
miQC
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/miQC.html" class="external-link">Running miQC on SRTpipeline objects</a>
</td>
<td style="text-align:left;">
<a href="https://www.biorxiv.org/content/10.1101/2021.03.03.433798v1" class="external-link">Hippen et. al., bioRxiv 2021</a>
</td>
<td style="text-align:left;">
<a href="https://github.com/greenelab/miQC" class="external-link">https://github.com/greenelab/miQC</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
tricycle
</td>
<td style="text-align:left;">
<a href="http://htmlpreview.github.io/?https://github.com/satijalab/SRTpipeline-wrappers/blob/master/docs/tricycle.html" class="external-link">Running estimate_cycle_position from tricycle on SRTpipeline Objects</a>
</td>
<td style="text-align:left;">
<a href="https://doi.org/10.1101/2021.04.06.438463" class="external-link">Zheng et. al., bioRxiv 2021</a>
</td>
<td style="text-align:left;">
<a href="https://www.bioconductor.org/packages/release/bioc/html/tricycle.html" class="external-link">https://www.bioconductor.org/packages/release/bioc/html/tricycle.html</a>
</td>
</tr>
</tbody>
</table>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
