<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A basic overview of single data analysis • SRTpipeline</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="A basic overview of single data analysis">
<meta property="og:description" content="SRTpipeline">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SRTpipeline</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="../articles/Installation.html">
    <span class="fas fa-book"></span>
     
    Install
  </a>
</li>
<li>
  <a href="../articles/get_started.html">
    <span class="fas fa-book"></span>
     
    Get Started
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book"></span>
     
    Single Data Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/single_basic_tutorial.html">Basic tutorial</a>
    </li>
    <li>
      <a href="../articles/dlpfc_tutorial.html">Embedding and clustering introduction</a>
    </li>
    <li>
      <a href="../articles/single_Embryo_SC_tutorial.html">SRT data clustering</a>
    </li>
    <li>
      <a href="../articles/single_Embryo_DRSC_tutorial.html">SRT data embedding and clustering</a>
    </li>
    <li>
      <a href="../articles/single_MOB_Anno_tutorial.html">SRT data embedding and annotation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book"></span>
     
    Multiple Data Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/multibatch_basic_tutorial.html">Basic tutorial</a>
    </li>
    <li>
      <a href="../articles/integration_introduction.html">Basic tutorial v2</a>
    </li>
    <li>
      <a href="../articles/multibatch_brain12_tutorial.html">SRT data clustering and integration</a>
    </li>
    <li>
      <a href="../articles/multibatch_HCC4_tutorial.html">SRT data embedding, clustering and integration example 1</a>
    </li>
    <li>
      <a href="../articles/multibatch_liver8_tutorial.html">SRT data embedding, clustering and integration example 2</a>
    </li>
    <li>
      <a href="../articles/multibatch_MOB16_tutorial.html">SRT data embedding, clustering and integration example 3</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/faq.html">
    <span class="fa fa-question-circle-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>A basic overview of single data analysis</h1>
            
            <h4 data-toc-skip class="date">Compiled: 十二月 06, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/feiyoung/SRTpipeline/blob/HEAD/vignettes/single_basic_tutorial.Rmd"><code>vignettes/single_basic_tutorial.Rmd</code></a></small>
      <div class="hidden name"><code>single_basic_tutorial.Rmd</code></div>

    </div>

    
    
<hr>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This tutorial demonstrates how to use <code>SRTpipeline</code> (&gt;=0.1.0) to analyze a spatially-resolved transcriptomics (SRT) data from 10X Visium platform. This tutorial will cover the following tasks, which we believe will be common for many spatial analyses:</p>
<ul>
<li>Normalization</li>
<li>Feature selection, including detecting spatially-variable features</li>
<li>Dimensional reduction</li>
<li>Clustering</li>
<li>Interactive visualization</li>
<li>Integration with single-cell RNA-seq data</li>
<li>Working with multiple slices</li>
</ul>
</div>
<div class="section level2">
<h2 id="prepare-srtproject">Prepare SRTProject<a class="anchor" aria-label="anchor" href="#prepare-srtproject"></a>
</h2>
<p>For this tutorial, we will introduce how to create a <code>SRTProject</code> object with single SRT sample using SRTpipeline that includes an introduction to common analytical workflows for a single data batch. Here, we will be taking a spatial transcriptomics dataset (SampleID: 151672) for human dorsolateral prefrontal cortex (DLPFC) as an example. There are 4015 spots and 33538 genes that were sequenced on the 10x Visium platform. Our preprocessed data can be downloaded <a href="https://github.com/feiyoung/PRECAST/blob/main/vignettes_data/dlpfc_151672.rda?raw=true" class="external-link">here</a>, and the raw data can be found <a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">here</a>.</p>
<p>First, we loaded the required packages with the command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/feiyoung/SRTpipeline">SRTpipeline</a></span><span class="op">)</span></code></pre></div>
<p>Next, we download the data to the current working path for the followed analysis by the following command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/feiyoung/PRECAST/blob/main/vignettes_data/dlpfc_151672.rda?raw=true"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>, <span class="st">"dlpfc_151672.rda"</span>, mode <span class="op">=</span> <span class="st">"wb"</span><span class="op">)</span></code></pre></div>
<p>Then load to R. The data is saved in a Seurat object named <code>dlpfc_151672</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"dlpfc_151672.rda"</span><span class="op">)</span></code></pre></div>
<details><p><summary><strong>How to read the raw sptial transcriptomics data?</strong></summary> In addition, if users’ data are in the raw form from spaceranger. We start by reading in the data. The <code><a href="https://satijalab.org/seurat/reference/Read10X.html" class="external-link">Read10X()</a></code> function reads in the output of the <a href="https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/what-is-space-ranger" class="external-link">spaceranger</a> pipeline from 10X, returning a unique molecular identified (UMI) count matrix and a spatial coordinates file. The values in count matrix represent the number of molecules for each feature (i.e. gene; row) that are detected in each spot (column). The spatial information of all spots is represented by the spatial coordinates file.</p>
We next use the count matrix to create a <code>Seurat</code> object. The object serves as a container that contains both data (like the count matrix, spatial coordinates) and analysis (like PCA, or clustering results) for a spatial transcriptomics dataset. For a technical discussion of the <code>Seurat</code> object structure, check the <a href="https://github.com/satijalab/seurat/wiki" class="external-link">GitHub Wiki</a>. For example, the count matrix is stored in <code>pbmc[["RNA"]]@counts</code>.
</details><details><p><summary><strong>What does data in a count matrix look like?</strong></summary></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Let's examine a few genes in the first five spots</span>
<span class="va">count</span> <span class="op">&lt;-</span> <span class="va">dlpfc_151672</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">counts</span>
<span class="va">count</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></code></pre></div>
<p>The <code>.</code> values in the matrix represent 0s (no molecules detected). Since most values in an count matrix are 0, a sparse-matrix representation is necessary to save memory and speed savings for data.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta_data</span> <span class="op">&lt;-</span> <span class="va">dlpfc_151672</span><span class="op">@</span><span class="va">meta.data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">meta_data</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></code></pre></div>
</details><p>We check the data by printing it. We can see there are 4015 spots and 33538 genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dlpfc_151672</span></code></pre></div>
<p>Check the meta data: each row contains the meta information of a spot with row name such as <code>AAACAAGTATCTCCCA-1</code>, which consists of the barcode, to identify each spot in RNA sequencing, separateor <code>-</code> and a number <code>1</code>; note that the columns, <code>row</code>, <code>col</code>, <code>imagerow</code> and <code>imagecol</code>, save the spatial information of each spot, in wihch <code>row</code> and <code>col</code> are the location of a spot on the slide and will be used in DR-SC model, <code>imagerow</code> and <code>imagecol</code> are the location of a spot on the H&amp;E image; there is the manual annotations <code>layer_guess_reordered</code> based on the cytoarchitecture in the original study (Maynard et al., 2021).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dlpfc_151672</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></code></pre></div>
<p>Check the manual annotations: this slice is annotated as five layers: Layer 3-6 and White matter.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">dlpfc_151672</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">layer_guess_reordered</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pre-processing-workflow">Pre-processing workflow<a class="anchor" aria-label="anchor" href="#pre-processing-workflow"></a>
</h2>
<div class="section level3">
<h3 id="create-a-srtproject-object">Create a SRTProject object<a class="anchor" aria-label="anchor" href="#create-a-srtproject-object"></a>
</h3>
<p>First, we show how to create a SRTProject object step by step.</p>
<p>Because the gene name is ensembl in this data, we change the genes’ ensembl name to symbols for convenience of followed analyses.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="va">dlpfc_151672</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">counts</span>
<span class="co">## Use eg.db database: this method is fast</span>
<span class="va">symbol_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transferGeneNames.html">transferGeneNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span>, now_name <span class="op">=</span> <span class="st">"ensembl"</span>, to_name <span class="op">=</span> <span class="st">"symbol"</span>,
    species <span class="op">=</span> <span class="st">"Human"</span>, Method <span class="op">=</span> <span class="st">"eg.db"</span><span class="op">)</span>
<span class="va">symbol_name</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">symbol_name</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## create count matrix list: note each component has a name, i.e., `ID151672`.</span>

<span class="va">cntList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>ID151672 <span class="op">=</span> <span class="va">count_matrix</span><span class="op">)</span>

<span class="co">## create spatial coordinate matrix</span>
<span class="va">coordList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">dlpfc_151672</span><span class="op">$</span><span class="va">row</span>, <span class="va">dlpfc_151672</span><span class="op">$</span><span class="va">col</span><span class="op">)</span><span class="op">)</span>

<span class="co">## create metadata list</span>
<span class="va">meta.data</span> <span class="op">&lt;-</span> <span class="va">dlpfc_151672</span><span class="op">@</span><span class="va">meta.data</span>
<span class="va">metadataList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">meta.data</span><span class="op">)</span>

<span class="co">## create meta data for each data batches. Here we only have one data batch.</span>
<span class="va">sampleMetadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Human"</span>, tissues <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DLPFC"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">sampleMetadata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cntList</span><span class="op">)</span>

<span class="co">## Name of this project</span>
<span class="va">projectName</span> <span class="op">&lt;-</span> <span class="st">"DLPFC151672"</span>

<span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">dlpfc_151672</span><span class="op">)</span></code></pre></div>
<p>Next, we start creating SRTProject object. We can print the basic information of this object, including three parts. The first part have the class of this object, outputPath of data that require to output, h5filePath that save the memory-cusuming data (i.e., count, logcount, …). The second part is about the datasets basic information, such as how many data batches(sample) and the data names, sample meta data (sampleColData) and meta data for each spot (cellMetaData). The last part is about downstream analyses information that is empty when this object created.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CreateSRTProject.html">CreateSRTProject</a></span><span class="op">(</span><span class="va">cntList</span>, <span class="va">coordList</span>, projectName <span class="op">=</span> <span class="va">projectName</span>, <span class="va">metadataList</span>, <span class="va">sampleMetadata</span>,
    force <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">SRTProj</span></code></pre></div>
<details><p><summary><strong>What does data in a h5 file look like?</strong></summary></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">h5ls</span><span class="op">(</span><span class="va">SRTProj</span><span class="op">@</span><span class="va">projectMetadata</span><span class="op">$</span><span class="va">h5filePath</span><span class="op">)</span></code></pre></div>
</details>
</div>
<div class="section level3">
<h3 id="normalizing-the-data">Normalizing the data<a class="anchor" aria-label="anchor" href="#normalizing-the-data"></a>
</h3>
<p>After removing unwanted cells and genes from the dataset, the next step is to normalize the data. To save RAM memory, normalized values are stored in disk as a <code>h5file</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeSRT.html">normalizeSRT</a></span><span class="op">(</span><span class="va">SRTProj</span><span class="op">)</span></code></pre></div>
<div class="section level4">
<h4 id="feature-selection">Feature selection<a class="anchor" aria-label="anchor" href="#feature-selection"></a>
</h4>
<p>We next select a subset of genes that exhibit high spot-to-spot variation in the dataset (i.e, they are highly expressed in some spots, and lowly expressed in others). It has been found that focusing on these genes in downstream analysis helps to highlight biological signal in single-cell datasets <a href="https://www.nature.com/articles/nmeth.2645" class="external-link">here</a>.</p>
<p>Then we choose variable features. The default number of variable features is 2,000, and users can change it using argument <code>nfeatures</code>. The default type is highly variable genes (HVGs), but users can use spatially variable genes by seting <code>type='SVGs'</code>, then <code>SPARK-X</code> will be used to choose SVGs. #### Choose highly variable features We can also directly use <code><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures()</a></code> to choose the top highly variable genes (HVGs). We found the performance of using SVGs or HVGs has no significant difference for the downstream analyses, such as dimension reduction and spatial clustering.</p>
<div class="section level5">
<h5 id="choose-spatially-variable-features">Choose spatially variable features<a class="anchor" aria-label="anchor" href="#choose-spatially-variable-features"></a>
</h5>
<p>We first applied the SPARK-X to choose the top spatially variable genes (SVGs). SPARK-X is embeded in the <code>DR.SC</code> R package and is implement in the <code><a href="https://rdrr.io/pkg/DR.SC/man/FindSVGs.html" class="external-link">FindSVGs()</a></code> function. By default, we return 2,000 genes per dataset. These will be used in downstream analysis, like probabilistic embeddings and spatial clustering. To further speed up the computation, we first use <code><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures()</a></code> to select 5,000 highly variable genes, then use SPARK-X to choose top SVGs in the implementation of <code><a href="https://rdrr.io/pkg/DR.SC/man/FindSVGs.html" class="external-link">FindSVGs()</a></code> function. The selected SVGs are saved in <code>dlpfc@assays$RNA@var.features</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectVariableFeatures.html">selectVariableFeatures</a></span><span class="op">(</span><span class="va">SRTProj</span><span class="op">)</span></code></pre></div>
<p>Calculate the adjcence matrix</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Obtain adjacence matrix</span>
<span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddAdj.html">AddAdj</a></span><span class="op">(</span><span class="va">SRTProj</span>, platform <span class="op">=</span> <span class="st">"Visium"</span>, force <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p>Visualize</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">selectFeatures</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">SRTProj</span><span class="op">@</span><span class="va">geneMetaData</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">SRTProj</span><span class="op">@</span><span class="va">geneMetaData</span><span class="op">$</span><span class="va">`isVGs#combine`</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>
<span class="fu">exprSpaHeatMap</span><span class="op">(</span><span class="va">SRTProj</span>, features <span class="op">=</span> <span class="va">selectFeatures</span>, title_name <span class="op">=</span> <span class="cn">T</span>, quantVec <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.9</span>, <span class="fl">0.9</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="dimension-reduction">Dimension reduction<a class="anchor" aria-label="anchor" href="#dimension-reduction"></a>
</h2>
<p>PCA is the most popular dimension reduction technique in single cell RNA sequencing (scRNA-seq) data and SRT　data because of its simplicity, computational efficiency, and relatively comparable performance. SRTpipeline provided three versions of PCA: standard PCA (PCA), approximated PCA (APCA), and weighted PCA (WPCA) by the function <code>AddPCA</code> with default version as APCA for fastest computation. The 15-dimensional PCs are extracted by default, and users can use their own values.</p>
<p>After running <code>AddPCA</code>, we would see the output of <code>SRTProj</code> includes <code>PCA</code> in the <code>Low-dimensional embeddings</code> field.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## APCA</span>
<span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddPCA.html">AddPCA</a></span><span class="op">(</span><span class="va">SRTProj</span>, n_comp <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>
<span class="co"># accurate PCA</span>

<span class="co"># SRTProj &lt;- AddPCA(SRTProj, Method='PCA')</span>

<span class="co"># weighted PCA</span>

<span class="co"># SRTProj &lt;- AddPCA(SRTProj, Method='WPCA')</span>
<span class="va">SRTProj</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="clustering-using-sc-meb">Clustering using SC-MEB<a class="anchor" aria-label="anchor" href="#clustering-using-sc-meb"></a>
</h2>
<p>Some SRT clustering methods use Markov random field to model clusters of spots, such as SC-MEB and DR-SC. These approaches work extremely well for data with spatial smoothness and are a standard practice in SRT data. For this reason, SRTpipeline uses existing state-of-the-art clustering methods from SRT packages for clustering.</p>
<p>We have had the most success using the Markov random field model implemented by SC-MEB. In SRTpipeline, SC-MEB clustering is performed using the <code><a href="../reference/Cluster_SCMEB.html">Cluster_SCMEB()</a></code> function.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Cluster_SCMEB.html">Cluster_SCMEB</a></span><span class="op">(</span><span class="va">SRTProj</span>, K <span class="op">=</span> <span class="fl">5</span>, reduction <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span>
<span class="va">SRTProj</span></code></pre></div>
<div class="section level3">
<h3 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h3>
<p>To run tSNE in SRTpipeline we use the AddTSNE() function:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddTSNE.html">AddTSNE</a></span><span class="op">(</span><span class="va">SRTProj</span>, n_comp <span class="op">=</span> <span class="fl">2</span>, reduction <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span></code></pre></div>
<p>The <code>reduction</code> tells <code><a href="../reference/AddTSNE.html">AddTSNE()</a></code> function uses specified reduction in the <code>SRTProj@reductions</code> slot.</p>
<p>To plot the two-dimensional tSNE results, we use the <code><a href="../reference/EmbedPlot.html">EmbedPlot()</a></code> function and pass the name of the tSNE embedding we just generated (“tSNE”). We can tell SRTpipeline how to color the spots by using <code>item</code> argument which tells SRTpipeline which matrix to use to find the specified metadata column provided to name.</p>
<p><code>SRTpipeline</code> package oﬀers a variety of visualization tools. We visualized the inferred domain types on embeddings using two components from tSNE. The tSNE plot showed the domain clusters were well segregated.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_pca_tsne2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EmbedPlot.html">EmbedPlot</a></span><span class="op">(</span><span class="va">SRTProj</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, plotEmbeddings <span class="op">=</span> <span class="st">"tSNE"</span>, cols <span class="op">=</span> <span class="fu"><a href="../reference/chooseColors.html">chooseColors</a></span><span class="op">(</span>n_colors <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,
    legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>
<span class="va">p_pca_tsne2</span></code></pre></div>
<p>To run UMAP in SRTpipeline we use the AddUMAP() function. Frist, we evaluate the two-dimensional UMAPs.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddUMAP.html">AddUMAP</a></span><span class="op">(</span><span class="va">SRTProj</span>, n_comp <span class="op">=</span> <span class="fl">2</span>, reduction <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span>
<span class="va">p_pca_umap2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EmbedPlot.html">EmbedPlot</a></span><span class="op">(</span><span class="va">SRTProj</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, plotEmbeddings <span class="op">=</span> <span class="st">"UMAP"</span>, cols <span class="op">=</span> <span class="fu"><a href="../reference/chooseColors.html">chooseColors</a></span><span class="op">(</span>n_colors <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,
    legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>
<span class="va">p_pca_umap2</span></code></pre></div>
<p>Next, we merge all plots into one figure.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span>
<span class="va">p_all</span> <span class="op">&lt;-</span> <span class="va">p_pca_tsne2</span> <span class="op">+</span> <span class="va">p_pca_umap2</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>nrow <span class="op">=</span> <span class="fl">1</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="va">p_all</span></code></pre></div>
<p>To save the plot, we can use <code><a href="../reference/write_fig.html">write_fig()</a></code> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/write_fig.html">write_fig</a></span><span class="op">(</span><span class="va">p_all</span>, filename <span class="op">=</span> <span class="st">"SCMEB_plots.png"</span><span class="op">)</span></code></pre></div>
<p>Except for the embedding plots, SRTpipeline also provides a variaty of visualization functions. First, we visualize the spatial distribution of cluster labels that shows the layer structure.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## choose colors to function chooseColors</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chooseColors.html">chooseColors</a></span><span class="op">(</span>n_colors <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="fu"><a href="../reference/EachClusterSpaHeatMap.html">EachClusterSpaHeatMap</a></span><span class="op">(</span><span class="va">SRTProj</span>, cols <span class="op">=</span> <span class="va">cols</span>, legend.position <span class="op">=</span> <span class="st">"right"</span>, base_size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>
<span class="co"># remove the border EachClusterSpaHeatMap(SRTProj, cols=cols,</span>
<span class="co"># legend.position='right',base_size=16, border_col='white')</span></code></pre></div>
<p>We plotted the heatmap of Pearson’s correlation coefcients of the PCA embeddings among the detected domains shows the good separation of the estimated embeddings across domains and the correlations between deeper layers were high, e.g., there were high correlations between domain 2 and 3, while correlations among the separated layers were low, i.e., domain 1 and 4.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/CCHeatMap.html">CCHeatMap</a></span><span class="op">(</span><span class="va">SRTProj</span>, reduction <span class="op">=</span> <span class="st">"PCA"</span>, grp_color <span class="op">=</span> <span class="va">cols</span><span class="op">)</span></code></pre></div>
<p>After adding the quantities for data visualization, the <code>SRTProject</code> object will have more information in the downstream analyses information. Now, we print this <code>SRTProject</code> object to check it. We observed two components added in the slot <code>plotEmbeddings</code> (Embeddings for plotting): <code>tSNE</code>, <code>tSNE3</code>, <code>UMAP</code> and <code>UMAP3</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SRTProj</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="deg-analysis">DEG analysis<a class="anchor" aria-label="anchor" href="#deg-analysis"></a>
</h2>
<p>To do downstream analyses, we require to get the count data from h5file. The function <code><a href="../reference/getGeneSpotData.html">getGeneSpotData()</a></code> access the gene-by-spot matrix and other data in <code>SRTProj</code> and then return a <code>SpatialExperiment</code> object, including two assays: counts, logcounts; rowData: from <code>SRTProj@geneMetaData</code>; colData: from <code>SRTProj@cellMetaData</code>, <code>SRTProj@clusters</code> and sample_id; reducedDim: from <code>SRTProj@reductions</code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGeneSpotData.html">getGeneSpotData</a></span><span class="op">(</span><span class="va">SRTProj</span><span class="op">)</span>
<span class="va">spe</span></code></pre></div>
<p>After obtain the spatial cluster labels using a clustering model, we can perform differentially expression analysis. First, we detect the DE genes for cluster 3.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat_deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindDEGs.html">FindDEGs</a></span><span class="op">(</span><span class="va">spe</span>, cluster1 <span class="op">=</span> <span class="st">"3"</span><span class="op">)</span>
<span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Seurat-methods.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">dat_deg</span>, <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">0.25</span><span class="op">)</span></code></pre></div>
<p>We perform differential expression analysis for all clusters by using <code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers()</a></code> function, then the DE genes’ information is saved in a data.frame object <code>dat_degs</code>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat_degs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindAllDEGs.html">FindAllDEGs</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span>
<span class="va">dat_degs</span></code></pre></div>
<p>We identify the significant DE genes by two criteria: (a) adjustd p-value less than 0.01 and (b) average log fold change greater than 0.4.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">degs_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Seurat-methods.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">dat_degs</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.01</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">0.25</span><span class="op">)</span>
<span class="va">degs_sig</span></code></pre></div>
<p>In the following, we perform gene set enrichment analysis for the DE genes of each Domain identified by DR-SC model using R package <code>gprofiler2</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gprofiler2</span><span class="op">)</span>
<span class="va">termList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">k</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span> <span class="op">{</span>
    <span class="co"># k &lt;- 1</span>
    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"k = "</span>, <span class="va">k</span>, <span class="st">"\n"</span><span class="op">)</span>
    <span class="va">dat_degs_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Seurat-methods.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">degs_sig</span>, <span class="va">cluster</span> <span class="op">==</span> <span class="va">k</span><span class="op">)</span>

    <span class="va">que1</span> <span class="op">&lt;-</span> <span class="va">dat_degs_sub</span><span class="op">$</span><span class="va">gene</span>
    <span class="va">gostres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gost</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">que1</span>, organism <span class="op">=</span> <span class="st">"hsapiens"</span>, correction_method <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span>
    <span class="va">termList</span><span class="op">[[</span><span class="va">k</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">gostres</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">termList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">result</span><span class="op">)</span></code></pre></div>
<p>To understand the functions of the identified spatial domains by DR-SC model, we compare the top significant biological process (BP) pathways in GO database for the DE genes from Domain 1 and 2. Here, we only show to visualize the significant BP pathways and users can explore the other databases such as KEGG and HPA.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Most commonly used databases</span>
<span class="va">source_set</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:BP"</span>, <span class="st">"GO:CC"</span>, <span class="st">"GO:MF"</span>, <span class="st">"KEGG"</span>, <span class="st">"HPA"</span><span class="op">)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"steelblue3"</span>, <span class="st">"goldenrod"</span>, <span class="st">"brown3"</span>, <span class="st">"#f98866"</span>, <span class="st">"#CE6DBD"</span><span class="op">)</span>
<span class="co">## Here, we show GO:BP</span>
<span class="va">source1</span> <span class="op">&lt;-</span> <span class="st">"GO:BP"</span>
<span class="va">ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">source_set</span> <span class="op">==</span> <span class="va">source1</span><span class="op">)</span>
<span class="va">ntop</span> <span class="op">=</span> <span class="fl">5</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cols</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">source_set</span>
<span class="va">pList_enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">ii</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span> <span class="op">{</span>
    <span class="co">## ii &lt;- 3</span>
    <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"ii="</span>, <span class="va">ii</span><span class="op">)</span>
    <span class="va">gostres2</span> <span class="op">&lt;-</span> <span class="va">termList</span><span class="op">[[</span><span class="va">ii</span><span class="op">]</span><span class="op">]</span>
    <span class="va">dat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Seurat-methods.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">gostres2</span><span class="op">$</span><span class="va">result</span>, <span class="va">term_size</span> <span class="op">&lt;</span> <span class="fl">500</span><span class="op">)</span>
    <span class="va">dat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_top_pathway.html">get_top_pathway</a></span><span class="op">(</span><span class="va">dat1</span>, ntop <span class="op">=</span> <span class="va">ntop</span>, source_set <span class="op">=</span> <span class="va">source1</span><span class="op">)</span>
    <span class="va">dat1</span> <span class="op">&lt;-</span> <span class="va">dat1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">)</span>, <span class="op">]</span>
    <span class="va">dat1</span><span class="op">$</span><span class="va">nlog10P</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">$</span><span class="va">p_value</span><span class="op">)</span>

    <span class="va">pList_enrich</span><span class="op">[[</span><span class="va">ii</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/barPlot_enrich.html">barPlot_enrich</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">$</span><span class="va">nlog10P</span><span class="op">)</span>, <span class="op">]</span>, source <span class="op">=</span> <span class="st">"source"</span>, <span class="st">"term_name"</span>,
        <span class="st">"nlog10P"</span>, cols <span class="op">=</span> <span class="va">cols</span><span class="op">[</span><span class="va">source_set</span><span class="op">[</span><span class="va">ss</span><span class="op">]</span><span class="op">]</span>, base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"-log10(p-adj)"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Biological terms"</span><span class="op">)</span> <span class="op">+</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Domain"</span>, <span class="va">ii</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span>
<span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">pList_enrich</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">]</span>, nrow <span class="op">=</span> <span class="fl">2</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p>We take out the top DE genes for each cluster for visualization.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">3</span>
<span class="va">dat_degs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="va">as.data.frame</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="va">n</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">topGene</span>
<span class="va">topGene</span></code></pre></div>
<p>We visualize the DE genes for each cluster group by gene-by-cell heatmap using the <code><a href="../reference/GCHeatMap.html">GCHeatMap()</a></code> function.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cols_cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/chooseColors.html">chooseColors</a></span><span class="op">(</span>n_colors <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EachGCHeatMap.html">EachGCHeatMap</a></span><span class="op">(</span><span class="va">spe</span>, features <span class="op">=</span> <span class="va">topGene</span><span class="op">$</span><span class="va">gene</span>, grp_color <span class="op">=</span> <span class="va">cols_cluster</span>, y_text_size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>
<span class="va">p1</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="trajectory-inference">Trajectory inference<a class="anchor" aria-label="anchor" href="#trajectory-inference"></a>
</h2>
<p>Next, we performed trajectory inference using the PCA embeddings and domain labels estimated by SC-MEB model. The <code><a href="../reference/EmbedPlot.html">EmbedPlot()</a></code> function can be used to visualize the inferred pseudotime on a specified embedding. If there is only one data batch, the function <code><a href="../reference/EachEmbedPlot.html">EachEmbedPlot()</a></code> plays the same role as function <code><a href="../reference/EmbedPlot.html">EmbedPlot()</a></code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddTrajectory.html">AddTrajectory</a></span><span class="op">(</span><span class="va">spe</span>, reduction <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EmbedPlot.html">EmbedPlot</a></span><span class="op">(</span><span class="va">spe</span>, reduction <span class="op">=</span> <span class="st">"PCA"</span>, colour_by <span class="op">=</span> <span class="st">"PT"</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EmbedPlot.html">EmbedPlot</a></span><span class="op">(</span><span class="va">spe</span>, reduction <span class="op">=</span> <span class="st">"Coord"</span>, colour_by <span class="op">=</span> <span class="st">"PT"</span><span class="op">)</span>
<span class="fu"><a href="../reference/drawFigs.html">drawFigs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span><span class="op">)</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, common.legend <span class="op">=</span> <span class="cn">TRUE</span>, legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>
<span class="co"># EachEmbedPlot(spe, reduction = 'PCA', colour_by='TrajPT') save(spe, SRTProj,</span>
<span class="co"># file=paste0(SRTProj@projectMetadata$outputPath,'/SRTProj.rds')) load('F:/Research</span>
<span class="co"># paper/IntegrateDRcluster/AnalysisCode/SRTpipeline/vignettes/DLPFC151672/SRTProj.rds')</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="other-downstream-analyses">Other downstream analyses<a class="anchor" aria-label="anchor" href="#other-downstream-analyses"></a>
</h2>
<details><p><summary><strong>Session Info</strong></summary></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">## Running under: Windows 10 x64 (build 22000)</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] LC_COLLATE=Chinese (Simplified)_China.936 </span>
<span class="co">## [2] LC_CTYPE=Chinese (Simplified)_China.936   </span>
<span class="co">## [3] LC_MONETARY=Chinese (Simplified)_China.936</span>
<span class="co">## [4] LC_NUMERIC=C                              </span>
<span class="co">## [5] LC_TIME=Chinese (Simplified)_China.936    </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] SRTpipeline_0.1.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2                  reticulate_1.25            </span>
<span class="co">##   [3] tidyselect_1.1.2            htmlwidgets_1.5.4          </span>
<span class="co">##   [5] grid_4.1.2                  BiocParallel_1.28.3        </span>
<span class="co">##   [7] Rtsne_0.16                  munsell_0.5.0              </span>
<span class="co">##   [9] ScaledMatrix_1.2.0          codetools_0.2-18           </span>
<span class="co">##  [11] ragg_1.2.2                  ica_1.0-2                  </span>
<span class="co">##  [13] future_1.26.1               miniUI_0.1.1.1             </span>
<span class="co">##  [15] spatstat.random_2.2-0       colorspace_2.0-3           </span>
<span class="co">##  [17] progressr_0.10.1            Biobase_2.54.0             </span>
<span class="co">##  [19] knitr_1.37                  rstudioapi_0.13            </span>
<span class="co">##  [21] Seurat_4.1.1                stats4_4.1.2               </span>
<span class="co">##  [23] SingleCellExperiment_1.16.0 ROCR_1.0-11                </span>
<span class="co">##  [25] ggsignif_0.6.3              tensor_1.5                 </span>
<span class="co">##  [27] listenv_0.8.0               MatrixGenerics_1.6.0       </span>
<span class="co">##  [29] GenomeInfoDbData_1.2.7      polyclip_1.10-0            </span>
<span class="co">##  [31] rprojroot_2.0.3             parallelly_1.32.0          </span>
<span class="co">##  [33] vctrs_0.4.1                 generics_0.1.2             </span>
<span class="co">##  [35] xfun_0.29                   ggthemes_4.2.4             </span>
<span class="co">##  [37] R6_2.5.1                    GenomeInfoDb_1.30.1        </span>
<span class="co">##  [39] ggbeeswarm_0.6.0            rsvd_1.0.5                 </span>
<span class="co">##  [41] bitops_1.0-7                spatstat.utils_3.0-1       </span>
<span class="co">##  [43] cachem_1.0.6                DelayedArray_0.20.0        </span>
<span class="co">##  [45] assertthat_0.2.1            promises_1.2.0.1           </span>
<span class="co">##  [47] scales_1.2.0                beeswarm_0.4.0             </span>
<span class="co">##  [49] rgeos_0.5-9                 gtable_0.3.0               </span>
<span class="co">##  [51] beachmat_2.10.0             globals_0.15.0             </span>
<span class="co">##  [53] goftest_1.2-3               rlang_1.0.2                </span>
<span class="co">##  [55] systemfonts_1.0.4           splines_4.1.2              </span>
<span class="co">##  [57] rstatix_0.7.0               lazyeval_0.2.2             </span>
<span class="co">##  [59] broom_0.7.12                spatstat.geom_2.4-0        </span>
<span class="co">##  [61] yaml_2.3.6                  reshape2_1.4.4             </span>
<span class="co">##  [63] abind_1.4-5                 backports_1.4.1            </span>
<span class="co">##  [65] httpuv_1.6.5                tools_4.1.2                </span>
<span class="co">##  [67] ggplot2_3.3.6               ellipsis_0.3.2             </span>
<span class="co">##  [69] spatstat.core_2.4-4         jquerylib_0.1.4            </span>
<span class="co">##  [71] RColorBrewer_1.1-3          BiocGenerics_0.40.0        </span>
<span class="co">##  [73] ggridges_0.5.3              Rcpp_1.0.8.3               </span>
<span class="co">##  [75] plyr_1.8.7                  sparseMatrixStats_1.6.0    </span>
<span class="co">##  [77] zlibbioc_1.40.0             purrr_0.3.4                </span>
<span class="co">##  [79] RCurl_1.98-1.6              ggpubr_0.4.0               </span>
<span class="co">##  [81] rpart_4.1.16                deldir_1.0-6               </span>
<span class="co">##  [83] viridis_0.6.2               pbapply_1.5-0              </span>
<span class="co">##  [85] cowplot_1.1.1               S4Vectors_0.32.3           </span>
<span class="co">##  [87] zoo_1.8-10                  SeuratObject_4.1.0         </span>
<span class="co">##  [89] SummarizedExperiment_1.24.0 ggrepel_0.9.1              </span>
<span class="co">##  [91] cluster_2.1.2               fs_1.5.2                   </span>
<span class="co">##  [93] magrittr_2.0.3              GiRaF_1.0.1                </span>
<span class="co">##  [95] data.table_1.14.2           scattermore_0.8            </span>
<span class="co">##  [97] lmtest_0.9-40               RANN_2.6.1                 </span>
<span class="co">##  [99] fitdistrplus_1.1-8          matrixStats_0.62.0         </span>
<span class="co">## [101] PRECAST_1.3                 patchwork_1.1.1            </span>
<span class="co">## [103] mime_0.12                   evaluate_0.15              </span>
<span class="co">## [105] xtable_1.8-4                mclust_5.4.10              </span>
<span class="co">## [107] IRanges_2.28.0              gridExtra_2.3              </span>
<span class="co">## [109] compiler_4.1.2              scater_1.25.1              </span>
<span class="co">## [111] tibble_3.1.7                KernSmooth_2.23-20         </span>
<span class="co">## [113] crayon_1.5.1                htmltools_0.5.2            </span>
<span class="co">## [115] mgcv_1.8-39                 later_1.3.0                </span>
<span class="co">## [117] tidyr_1.2.0                 DBI_1.1.2                  </span>
<span class="co">## [119] formatR_1.11                MASS_7.3-55                </span>
<span class="co">## [121] car_3.0-12                  Matrix_1.4-0               </span>
<span class="co">## [123] cli_3.2.0                   parallel_4.1.2             </span>
<span class="co">## [125] igraph_1.3.2                DR.SC_3.0                  </span>
<span class="co">## [127] GenomicRanges_1.46.1        pkgconfig_2.0.3            </span>
<span class="co">## [129] pkgdown_2.0.6               sp_1.5-0                   </span>
<span class="co">## [131] plotly_4.10.0               scuttle_1.4.0              </span>
<span class="co">## [133] spatstat.sparse_2.1-1       vipor_0.4.5                </span>
<span class="co">## [135] bslib_0.3.1                 XVector_0.34.0             </span>
<span class="co">## [137] CompQuadForm_1.4.3          stringr_1.4.0              </span>
<span class="co">## [139] digest_0.6.29               sctransform_0.3.3          </span>
<span class="co">## [141] RcppAnnoy_0.0.19            spatstat.data_3.0-0        </span>
<span class="co">## [143] rmarkdown_2.11              leiden_0.4.2               </span>
<span class="co">## [145] uwot_0.1.11                 DelayedMatrixStats_1.16.0  </span>
<span class="co">## [147] shiny_1.7.1                 gtools_3.9.2.2             </span>
<span class="co">## [149] lifecycle_1.0.1             nlme_3.1-155               </span>
<span class="co">## [151] jsonlite_1.8.0              carData_3.0-5              </span>
<span class="co">## [153] BiocNeighbors_1.12.0        desc_1.4.0                 </span>
<span class="co">## [155] viridisLite_0.4.0           fansi_1.0.3                </span>
<span class="co">## [157] pillar_1.7.0                lattice_0.20-45            </span>
<span class="co">## [159] fastmap_1.1.0               httr_1.4.3                 </span>
<span class="co">## [161] survival_3.2-13             glue_1.6.2                 </span>
<span class="co">## [163] png_0.1-7                   stringi_1.7.6              </span>
<span class="co">## [165] sass_0.4.1                  textshaping_0.3.6          </span>
<span class="co">## [167] BiocSingular_1.10.0         memoise_2.0.1              </span>
<span class="co">## [169] dplyr_1.0.9                 irlba_2.3.5                </span>
<span class="co">## [171] future.apply_1.9.0</span></code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
